django_extensions/db/fields/__init__.py

```diff
@@ -143,9 +173,11 @@ def _slug_strip(self, value):
         value = re.sub('%s+' % re_sep, self.separator, value)
         return re.sub(r'^%s+|%s+$' % (re_sep, re_sep), '', value)
 
-    def slugify_func(self, content):
+    def slugify_func(self, content, model_instance):
         if content:
-            return self.slugify_function(content)
+            slugify_function = getattr(model_instance, 'slugify_function', None)
+            slugify_function = slugify_function or self.slugify_function
```

Comment:
Why not use:
```python
slugify_function = getattr(model_instance, 'slugify_function', self.slugify_function)
```

Mental note: wondering about the performance impact of doing the `getattr` here instead of in `slug_generator`, though I don't really like that alternative as well since I then much rather also refactor the `lambda` out...

Reply:
Let me have a look on what we could do in the outer lamda function :) Thanks for the thoughts

Reply:
- [x] Move getattr from ``slugify_func``
- [x] Added unit test for ``slugify_func``